git/repository/options.go

```diff
@@ -62,10 +62,16 @@ type CheckoutStrategy struct {
 	// Tag to checkout, takes precedence over Branch.
 	Tag string
 
-	// SemVer tag expression to checkout, takes precedence over Tag.
+	// SemVer tag expression to checkout, takes precedence over Branch and Tag.
 	SemVer string `json:"semver,omitempty"`
 
-	// Commit SHA1 to checkout, takes precedence over Tag and SemVer.
+	// RefName is the reference to checkout to. It must conform to the
+	// Git reference format: https://git-scm.com/book/en/v2/Git-Internals-Git-References
+	// Examples: "/refs/heads/main", "refs/pull/420/head", "refs/tags/v0.1.0"
```

Comment:
its the example thats wrong. i dont know what kind of validation we'd have for this. the error message contains the reference name so i think thats helpful enough for users to debug in case of a config errror.